x-pack/plugins/security_solution/public/common/components/discover_in_timeline/use_discover_in_timeline_actions.tsx

```diff
@@ -177,67 +180,99 @@ export const useDiscoverInTimelineActions = (
    * */
   const updateSavedSearch = useCallback(
     async (savedSearch: SavedSearch, timelineId: string) => {
-      dispatch(
-        startTimelineSaving({
-          id: timelineId,
-        })
-      );
       savedSearch.timeRestore = true;
       savedSearch.timeRange =
         savedSearch.timeRange ?? discoverDataService.query.timefilter.timefilter.getTime();
       savedSearch.tags = ['security-solution-default'];
 
+      // If there is already a saved search, only update the local state
       if (savedSearchId) {
         savedSearch.id = savedSearchId;
-      }
-      try {
-        const response = await persistSavedSearch(savedSearch, {
-          onTitleDuplicate: () => {},
-          copyOnSave: !savedSearchId,
-        });
-
-        if (!response || !response.id) {
-          throw new Error('Unknown Error occured');
+        if (!timelineRef.current.savedSearch) {
+          dispatch(
+            timelineActions.initializeSavedSearch({
+              id: TimelineId.active,
+              savedSearch,
+            })
+          );
```

Comment:
could early return approach possibly reduce the amount of nested conditions we have here?